pkg/k8s/apis/cilium.io/v2alpha1/bgp_cluster_types.go

```diff
@@ -0,0 +1,126 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package v2alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	slimv1 "github.com/cilium/cilium/pkg/k8s/slim/k8s/apis/meta/v1"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:resource:categories={cilium,ciliumbgp},singular="ciliumbgpclusterconfig",path="ciliumbgpclusterconfigs",scope="Cluster",shortName={bgpcluster}
+// +kubebuilder:printcolumn:JSONPath=".metadata.creationTimestamp",name="Age",type=date
+// +kubebuilder:storageversion
+
+// CiliumBGPClusterConfig is the Schema for the CiliumBGPClusterConfig API
+type CiliumBGPClusterConfig struct {
+	// +deepequal-gen=false
+	metav1.TypeMeta `json:",inline"`
+	// +deepequal-gen=false
+	metav1.ObjectMeta `json:"metadata"`
+
+	// Spec defines the desired cluster configuration of the BGP control plane.
+	Spec CiliumBGPClusterConfigSpec `json:"spec,omitempty"`
```

Comment:
`spec` should not be omitted from JSON serialization when it's a required field. Note that I see this issue with other APIs in this PR. If you change it here, do the same throughout the APIs.